### ISSUE TITLE
diff() returns -0 for the exact same 'start of' day 

### ISSUE THREAD

**Issue by bijanvakili** [moment/moment#2908]
`-0` (negative zero) seems to be expected behavior in certain cases (see issues #2503 and #2590).  The recommendation is to use `startOf('day')` when comparing dates based on the discussion in these tickets.

But if you run the following comparison using the exact same timestamp:
```javascript
var moment = require('moment-timezone');

function isNegative(n) {
  return (1 / n) < 0;
}

var ts = '1977-05-12 00:00:00.000';
var format = 'YYYY-MM-DD hh:mm:ss.SSS';
var timezone = 'America/Los_Angeles';

var d1 = moment(ts, format).tz(timezone).startOf('day');
var d2 = moment(ts, format).tz(timezone).startOf('day');

console.log('Diff on years is negative zero?: ' + isNegative(d2.diff(d1, 'years')));
console.log('Diff on months is negative zero?: ' + isNegative(d2.diff(d1, 'months')));
console.log('Diff on days is negative zero?: ' + isNegative(d2.diff(d1, 'days')));
```

The resulting output will be:
```
Diff on years is negative zero?: true
Diff on months is negative zero?: true
Diff on days is negative zero?: false
```

You can verify this using [this jsfiddle link](http://jsfiddle.net/1410pbev/).

It seems inconsistent that `years` and `months` are `-0` but `days` is regular `+0` for what should be the exact same point in time.

A few other things to note:
- I get the same result even if I don't use `moment-timezone`
- This does not occur for versions prior to `2.9.0` (e.g. version `2.8.4` does not have this issue)

I'm currently employing a workaround of explicitly adding `+0` to ensure I don't encounter `-0` as follows:
```javascript
console.log('Diff on years is negative zero?: ' + isNegative(d2.diff(d1, 'years') + 0));
```

If behavior is not a bug, would it be possible to update the `moment.js` documentation to provide an explanation and suggested workaround.

(P.S. Thanks for a great library!!)

**Comment by icambron** [comment]
It definitely looks like a bug, and thanks for the nice writeup. I think we'd accept a patch for this.

It sounds like the bug may be in moment-timezone? If so, it might make more sense to file the bug [there](https://github.com/moment/moment-timezone).

**Comment by maggiepint** [comment]
Fixed in #3059
